### DISCUSSION TITLE
Error message  by selection JSON-Protocol

### DISCUSSION THREAD

**Discussion by ardent200** [pkuehnel/TeslaSolarCharger#1634] [Q&A] [answered]
Hello, when I select the JSON protocol in the "Add Rest config" menu I get the error "An unhandled error has occurred. After pressing the "Reload" button the window closes. I have set up the same configuration on an old TSC version and updated again and again to the current version and it runs smoothly there. Now set up on a new Rasberry Pi with the current version of TSC and this error message appears, as described previously.

Can someone please help why it doesn't work  correctly. Thank you in advance.

Best Regards

Markus

![image](https://github.com/user-attachments/assets/7df32fce-666e-4846-8847-b4cc9235885e)

**Comment by pkuehnel** [comment]
Very strange. Try setting up everything without testing it, then you should be able so save the settings. That is definitely a bug.

**Reply by ardent200** [reply]
I have now saved it without testing, but no values ​​are output. After calling up “Configure” again and pressing the test button, the following message appears: "The request was canceled due to the configured HttpClient.Timeout of 1 seconds elapsing."
![image](https://github.com/user-attachments/assets/c040ae0f-1168-42f4-b304-384b41217fdc)
![image](https://github.com/user-attachments/assets/aee45466-0440-4eda-b73d-b3717f1da0ed)

**Reply by pkuehnel** [reply]
That means it takes longer than one second until TSC gets a response via the URL. Could that be the case?

**Reply by ardent200** [reply]
No, I don't think so because the same configuration runs on the "old" Rasberry Pi with Teslamate.
![image](https://github.com/user-attachments/assets/bccceb2f-183b-4de3-b85f-0c8a369f463d)

**Reply by pkuehnel** [reply]
Please upload your log files after you configured everything and it is not working: https://nextcloud.th-deg.de/s/2mT2QtRNfAgfXRD

**Reply by ardent200** [reply]
The Logfile is uploaded

**Reply by pkuehnel** [reply]
In TSC go to BASE CONFIGURATION -> scroll down to the bottom of the page and expand ADVANCED SETTINGS -> Compare SOLAR PLANT ADJUSTMENT INTERVAL on your two installations. The timeout is the same as the interval, so try increasing it.

**Reply by ardent200** [reply, marked as answer]
Thank you it works. I increased it on min. 3 second.
